cmd/ci-secret-bootstrap/main_test.go

```diff
@@ -2633,3 +2634,435 @@ func (f *fakeVaultClient) ListKVRecursively(prefix string) ([]string, error) {
 func (f *fakeVaultClient) UpsertKV(_ string, _ map[string]string) error {
 	return nil
 }
+
+func TestIntegration(t *testing.T) {
+	type vaultObject struct {
+		field string
+		data  []byte
+	}
+
+	testCases := []struct {
+		id              string
+		config          secretbootstrap.Config
+		secretGetters   map[string]coreclientset.SecretsGetter
+		vaultObjects    map[string][]vaultObject
```

Comment:
```suggestion
		vaultObjects    map[string]map[string]string
```

The data structure for the content of an item used by vault is a` map[string]string`, so making it a `[]struct{field string, data []byte}` introduces some potential for writing incorrect and hard to understand tests

Reply:
Yea this was only to control the initial vault data. I didn't do it with `map[string]string` because I wanted to control also the secret data as well.

Reply:
First off you can simply convert between `map[string]string` and `map[string][]byte` and second off, the important point is that the map value needs to be a map of some sort and not a slice. Making it a slice with implicit map restrictions just makes everything more complicated and error-prone

Reply:
Yea, my first thought was to create a map of a map but in order not to make things complicated I created the struct instead. I will make the changes. Thanks.